components/scream/src/share/tests/grid_tests.cpp

```diff
@@ -0,0 +1,123 @@
+#include <catch2/catch.hpp>
+
+#include "share/grid/simple_grid.hpp"
+#include "share/grid/se_grid.hpp"
+#include "share/grid/grids_manager.hpp"
+#include "share/grid/grid_utils.hpp"
+#include "share/scream_types.hpp"
+
+#include "ekat/ekat_pack.hpp"
+
+namespace {
+
+using namespace scream;
+using namespace scream::ShortFieldTagsNames;
+using device_type = DefaultDevice;
+using kokkos_types = KokkosTypes<device_type>;
+
+TEST_CASE("simple_grid", "") {
+
+  ekat::Comm comm(MPI_COMM_WORLD);
+  int num_procs = comm.size();
+  int num_cols = 128, num_levels = 72;
+  SimpleGrid grid("my_grid", num_cols, num_levels, comm);
+  REQUIRE(grid.type() == GridType::MeshFree);
+  REQUIRE(grid.name() == "my_grid");
+  REQUIRE(grid.get_num_vertical_levels() == num_levels);
+  REQUIRE(grid.get_num_local_dofs() == num_cols / num_procs);
+
+  auto lid_to_idx = grid.get_lid_to_idx_map();
+  for (int i = 0; i < grid.get_num_local_dofs(); ++i) {
+    REQUIRE(lid_to_idx.extent_int(1) == 1);
+    REQUIRE(i == lid_to_idx(i, 0));
```

Comment:
I suspect this will fail on GPU, since lid_to_idx is a device view. You need to deep copy to host first.